cap-primitives/src/std/fs/maybe_owned_file.rs

```diff
@@ -0,0 +1,22 @@
+use std::{fs, io};
+
+pub(crate) enum MaybeOwnedFile<'borrow> {
+    Owned(fs::File),
+    Borrowed(&'borrow fs::File),
+}
```

Comment:
Hmm, I'm wondering, could we somehow re-use the concept of clone-on-write smart pointer here? Just an idea. I'm not that familiar with `Cow` myself TBH, but it looks familiar.

Reply:
It is very similar; the difference is that `Cow` requires its `T` to implement `ToOwned`, which requires an infallible way to go from `&T` to `T`. `std::fs::File` can't do that because cloning a `File` is fallible.

Reply:
Ah, gotcha!

Reply:
If we ever find ourselves in the need of this fallible idiom, I guess we could extract it into a generic `TryCow<T: Fallible>` or some such. But until then, this looks good!

Reply:
I think we want something like `Cow` but without the whole "copy on write" thing going on ;-).